src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -87,6 +87,14 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
     @Optional
     final MapProperty<String, String> labels = project.objects.mapProperty(String, String)
 
+    /**
+     * The buildLabels are labels that will not invalidate the cache key when changed.
+     * Usually used to store in the image build information that changes all the time.
+     * They will only be applied when the image builds, otherwise they are ignored.
+     */
+    @Internal
+    final MapProperty<String, String> buildLabels = project.objects.mapProperty(String, String)
```

Comment:
I'm not sure how else to address this but it _could_ become confusing as folks may not know they can actually use it given the annotation.

Reply:
Can you **also** add an @input and @optional along with keeping the @internal ?

Reply:
The goal of this feature is 'not to be part of the inputs'
Let say you need some references in your image about the date it was build, the git id of the commit you used, stuff like that will change all the time and will invalidate the cache key all the time if part of the input.
My claim is that we need a property that enables us to save some information in the image when it is rebuilt, but it must not trigger the rebuild.

Reply:
A task property can only be ignored for up-to-date checking by not annotating it or annotating it with `@Internal`. It doesn't make sense to combine `@Internal` with any other input or output annotation.